src/pgapp_worker.erl

```diff
@@ -115,6 +96,17 @@ middle_man_transaction(Pool, Fun, Timeout) ->
             {error, Reason}
     end.
 
+wrap_timeout(Fun) ->
+    fun(Worker) ->
+            try
+                Fun(Worker)
+            catch
+                Type:Error ->
```

Comment:
So, are you sure you want to catch all errors, not `exit = Type:{timeout, _} = Reason`?